scanamo/src/main/scala/org/scanamo/ops/ScanamoOpsA.scala

```diff
@@ -1,48 +1,80 @@
 package org.scanamo.ops
 
+import cats.{Functor, Monad, MonadError}
 import com.amazonaws.services.dynamodbv2.model._
+import org.scanamo.ops.ScanamoOpsA.ScanamoResult
 import org.scanamo.request._
 
-sealed trait ScanamoOpsA[A] extends Product with Serializable
-final case class Put(req: ScanamoPutRequest) extends ScanamoOpsA[PutItemResult]
-final case class ConditionalPut(req: ScanamoPutRequest)
-    extends ScanamoOpsA[Either[ConditionalCheckFailedException, PutItemResult]]
-final case class Get(req: GetItemRequest) extends ScanamoOpsA[GetItemResult]
-final case class Delete(req: ScanamoDeleteRequest) extends ScanamoOpsA[DeleteItemResult]
-final case class ConditionalDelete(req: ScanamoDeleteRequest)
-    extends ScanamoOpsA[Either[ConditionalCheckFailedException, DeleteItemResult]]
-final case class Scan(req: ScanamoScanRequest) extends ScanamoOpsA[ScanResult]
-final case class Query(req: ScanamoQueryRequest) extends ScanamoOpsA[QueryResult]
-final case class BatchWrite(req: BatchWriteItemRequest) extends ScanamoOpsA[BatchWriteItemResult]
-final case class BatchGet(req: BatchGetItemRequest) extends ScanamoOpsA[BatchGetItemResult]
-final case class Update(req: ScanamoUpdateRequest) extends ScanamoOpsA[UpdateItemResult]
-final case class ConditionalUpdate(req: ScanamoUpdateRequest)
-    extends ScanamoOpsA[Either[ConditionalCheckFailedException, UpdateItemResult]]
+sealed trait ScanamoOpsA[+A] extends Product with Serializable
```

Comment:
I'm surprised about the variance annotation. Is it because of the `Either` in some ops?

Reply:
It was because of the `Nothing` in the `Fail` case